### ISSUE TITLE
Shooting Acid Through Water Cannon

### ISSUE THREAD

**Issue by Alm999** [CleverRaven/Cataclysm-DDA#78377]
### Is your feature request related to a problem? Please describe.

I have an idea. What if our survivors could use something more menacing for water cannon shooting, like hydrochloric acid (or any other acid, really)? It certainly fits into the "Be smart, choose your own terms for fight" motif of the game. We already can place any liquid into vehicle tanks, so why not go cosplaying spitter zombies ourselves?

This won't disrupt the balance because:
1) water cannons are fairly rare;
2) acids are not found in large enough quantities in the field and must be crafted (a single burst is still 1 L, so prepare to craft a lot);
3) normal acid resistances still apply for enemies and acid pools evaporate quickly so large amounts of acid is required in order to form a large enough pool of acid.

### Solution you would like.

Currently water cannons are not much more than "launchers" skill training devises capable only using water. Adding the ability to shoot acids would will make them more interesting. The immediate impact of the cannon's "shot" may have the same result as ordinary water (dealing 0 damage), but the spilled liquid would form a pool of acid and act as usual spilled acid (damaging whoever steps on it).

### Describe alternatives you have considered.

Maybe we could try to use other liquids as well? Like, shooting gasoline or ethanol to spread the fires (initial fire source shall be set separately because the water cannon does not have igniter). Or maybe beer, because… reasons? Maybe any liquid (non-volatile) shall be allowed to be used with the cannon, because it is certainly realistic and can potentially open new and interesting strategies.

**Comment by oosyrag** [comment]
The problem here is that in game water cannons are nonsense to begin with, and probably haven't been removed or reworked because they're rare enough that no one has cared. Fire truck hoses use pressure from the hydrant to "shoot" water.

Using them to train launchers skill is absurd too.

**Comment by Alm999** [comment]
> The problem here is that in game water cannons are nonsense to begin with, and probably haven't been removed or reworked because they're rare enough that no one has cared. Fire truck hoses use pressure from the hydrant to "shoot" water.

I disagree. IRL both fire engines and police anti-riot vehicles have water cannons and use them to quickly throw large (and I mean **large**, not just 1 L/s of liquid) amounts of water/chemicals. You can argue that our current implementation is a little bit off (the cannon does not seem to have a pump and does not require additional power to be operable), but the existence of such a weapon is justified. See:

https://armortek.com/water-cannon-riot-control-trucks

https://www.dreamstime.com/water-cannon-fire-engine-shooting-high-velocity-stream-water-firemen-fighting-fire-forest-water-cannon-fire-image194211307

Note there is no hydrant in sight. Stating that we should not have water cannons and the sole reason of their existence in the game is simply that nobody could be bothered to remove them is a little… bold.

> Using them to train launchers skill is absurd too.

I disagree x2. But we are not discussing skill training here and I do not wish to delve into game balance.

**Comment by RedMisao** [comment]
Sounds inconvenient. Acid is not that fast acting, not even HF, see this video for reference: https://www.youtube.com/watch?v=oipksRhISfM. Acid is bad for living things because it damages the protective layer of the organism (aka the skin in humans), plus they cause necrosis, and in the best case scenario they cause irritation. None of this applies to zombies.

Additionally, are you sure IRL water cannons are capable of shooting anything that's not water? Acids are denser and more viscuous than water, not to mention they could degrade their way through metal itself, depending on the acid and the alloy.

**Comment by Alm999** [comment]
> Sounds inconvenient. Acid is not that fast acting, not even HF, see this video for reference: https://www.youtube.com/watch?v=oipksRhISfM. Acid is bad for living things because it damages the protective layer of the organism (aka the skin in humans), plus they cause necrosis, and in the best case scenario they cause irritation. None of this applies to zombies.

Well, in-game acids work differently. I'm not against re-balancing acids so that their spills form any sources (zombie spitters/acidic zombies + their blood that spills out after smashing/butchering etc.) would not have any immediate effect (and not having any effect at all unless you are barefooted or knee-deep in it). Plus different acids have different effects IRL.

But right now in-game we have what we have: zombies spitting copious amounts of deadly (Alien™ level) acids while player can not throw even a single glass flask of acid to do the same (you can throw a bottle of acid but there is no guarantee it breaks on impact). Oh, and most zombies are *not* immune to zombie-acid. We can pump their own toxic blood from them, so why not throw it their way and let them have a taste of their own medicine?

> Additionally, are you sure IRL water cannons are capable of shooting anything that's not water? Acids are denser and more viscuous than water, not to mention they could degrade their way through metal itself, depending on the acid and the alloy.

Why not? It all comes down to pump strength.  If going absolutely realistic, we shall not store sulfuric acid in 200 L steel barrels (like it is on pump stations) because it is corrosive towards iron. But it is perfectly OK to store it there in-game. Currently we do not discriminate between acids and containers (anything can store anything).

Hydrochloric acid may be more viscous (https://www.hosecon.com/PDF/Engineering/Common/GatesViscocityData.pdf) but not that much, not honey-level of viscous. And its density is close to that of water. If this must be taken into consideration, then a simple maximum range adjustment (via range ammo bonus?) may be applied.

And as for longevity… It all comes down to the pump impeller material. Obviously cast iron will degrade quickly (but we do not know the material -- it could be Haselloy). But it is a life or death situation, not a matter of using the thing for years to come. 10 seconds of usage won't melt the thing (and ironically, concentrated acids are less corrosive than dilute ones, up to the point to aluminum alloys being usable with WFNA).
